R/read_from_drive.R

```diff
@@ -0,0 +1,53 @@
+#' @import Microsoft365R
+NULL
+
+#' Read a file from the SharePoint drive into R
+#'
+#' This function combines the `Microsoft365R::get_sharepoint_site()` function
+#' and the `rio::import()`` functions to connect to a sharepoint drive
+#' and download a file, then open it in R. Because it uses `rio::import()`,
+#' the function is able to open a large number of different filetypes.
+#'
+#' The defaults are set to open the main excel sheet for the eScience Center
+#' digital skills workshops (a.k.a. the "Holy Excel Sheet").
+#'
+#' @param path Path of the file inside the sharepoint drive
+#' @param drive URL of the sharepoint drive
+#'
+#' @return an imported R object
+#' @export
+read_from_drive <- function(path = "General/Digital Skills Workshops 2022.xlsx",
+                            drive = "https://nlesc.sharepoint.com/sites/instructors"){
+
+  #TODO validate path and drive arguments
+  # e.g. path needs to have an extension
+  # drive needs to be sharepoint site and include https://
+
+  # connect to drive
```

Comment:
Hi Barbara! Thanks a lot for optimizing this part :) I get the following error when trying to run `read_from_drive`:

```
ds_xlsx <- read_from_drive()

Loading Microsoft Graph login for default tenant
Error in login$get_sharepoint_site(site_url, site_id) : 
  object 'ms_site' not found
```

ms_site is a Microsoft365R function, so somewhere this is not imported :)

Reply:
this is now fixed